moto/xray/responses.py

```diff
@@ -0,0 +1,150 @@
+from __future__ import unicode_literals
+import json
+import datetime
+
+from moto.core.responses import BaseResponse
+from six.moves.urllib.parse import urlsplit
+
+from .models import xray_backends
+from .exceptions import AWSError, BadSegmentException
+
+
+class XRayResponse(BaseResponse):
+
+    def _error(self, code, message):
+        return json.dumps({'__type': code, 'message': message}), dict(status=400)
+
+    @property
+    def xray_backend(self):
+        return xray_backends[self.region]
+
+    @property
+    def request_params(self):
+        try:
+            return json.loads(self.body)
+        except ValueError:
+            return {}
+
+    def _get_param(self, param, default=None):
+        return self.request_params.get(param, default)
+
+    def _get_action(self):
```

Comment:
Nice work pulling out `_get_action`

Reply:
Yeah I had just reimplemented the call function but then it was bugging me any time I looked at that class :-)